platform/darwin/src/MGLStyle_Private.hpp

```diff
@@ -0,0 +1,11 @@
+#import "MGLStyle.h"
+
+#import "MGLStyleLayer.h"
+#import "MGLFillStyleLayer.h"
+#import <mbgl/util/default_styles.hpp>
+#include <mbgl/mbgl.hpp>
+
+@interface MGLStyle (Private)
+@property (nonatomic, weak) MGLMapView *mapView;
+@property (nonatomic) mbgl::Map *mbglMap;
```

Comment:
Good point. What if we just expose the `Map::update(Update flags)` method in a private header? I don't know about sources but we should be covered in terms of layers and paint properties.